src/OpenFeature.Contrib.Providers.Flagd/InProcessResolver.cs

```diff
@@ -0,0 +1,276 @@
+using Grpc.Net.Client;
+using OpenFeature.Flagd.Grpc;
+using System.Net.Sockets;
+using System.Threading;
+using Grpc.Core;
+using Value = OpenFeature.Model.Value;
+
+namespace OpenFeature.Contrib.Providers.Flagd
+{
+    internal class InProcessResolver : Resolver
+    {
+        static int EventStreamRetryBaseBackoff = 1;
+
+        private readonly FlagSyncService.FlagSyncServiceClient _client;
+        private readonly JsonEvaluator _evaluator;
+
+        private readonly Mutex _mtx;
+        private int _eventStreamRetries;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+        private readonly FlagdConfig _config;
+
+        internal InProcessResolver(FlagdConfig config)
+        {
+            _config = config;
+            _client = BuildClient(config, channel => new FlagSyncService.FlagSyncServiceClient(channel));
+            _mtx = new Mutex();
+            _evaluator = new JsonEvaluator(config.SourceSelector);
+        }
+
+        internal InProcessResolver(FlagSyncService.FlagSyncServiceClient client, FlagdConfig config)
+        {
+            _config = config;
+            _client = client;
+            _mtx = new Mutex();
+            _evaluator = new JsonEvaluator(config.SourceSelector);
+        }
+
+
+        public void Init()
+        {
+            var handleEvents = new Thread(HandleEvents);
+            handleEvents.Start();
+        }
+
+        public void Shutdown()
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            return Task.FromResult(_evaluator.ResolveBooleanValue(flagKey, defaultValue, context));
+        }
+
+        public Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+        {
+            return Task.FromResult(_evaluator.ResolveStringValue(flagKey, defaultValue, context));
+        }
+
+        public Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return Task.FromResult(_evaluator.ResolveIntegerValue(flagKey, defaultValue, context));
+        }
+
+        public Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return Task.FromResult(_evaluator.ResolveDoubleValue(flagKey, defaultValue, context));
+        }
+
+        public Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            return Task.FromResult(_evaluator.ResolveStructureValue(flagKey, defaultValue, context));
+        }
+
+        private async void HandleEvents()
+        {
+            while (_eventStreamRetries < _config.MaxEventStreamRetries)
```

Comment:
By default `while (_eventStreamRetries < _config.MaxEventStreamRetries)` evaluates to `False`, since `MaxEventStreamRetries` is zero (this means nothing is ever read). Additionally the retries ENV VAR seem to be ignored without the cache.